src/Microsoft.Kusto.ServiceLayer/DataSource/KustoClient.cs

```diff
@@ -167,21 +167,23 @@ public IDataReader ExecuteQuery(string query, CancellationToken cancellationToke
                         codeBlock.Service.GetMinimalText(MinimalTextKind.RemoveLeadingWhitespaceAndComments);
                     
                     IDataReader origReader;
-                
-                    if(minimalQuery.StartsWith(".") && !minimalQuery.StartsWith(".show")){
-                        origReader = _kustoAdminProvider.ExecuteControlCommand(
-                            KustoQueryUtils.IsClusterLevelQuery(minimalQuery) ? "" : databaseName,
-                            minimalQuery,
-                            clientRequestProperties);
-                    }
-                    else{
-                        origReader = _kustoQueryProvider.ExecuteQuery(
-                            KustoQueryUtils.IsClusterLevelQuery(minimalQuery) ? "" : databaseName,
-                            minimalQuery,
-                            clientRequestProperties);
-                    }
 
-                    origReaders[index] = origReader;
+                    if(!string.IsNullOrEmpty(minimalQuery)){        // Query is empty in case of comments
```

Comment:
For the empty DataReader in origReaders, does KustoResultsReader handle it correctly. I am assuming that it will be undefined/null and so KustoDataReader skips it. But please do check on this.

Reply:
Sending the entire text in query editor to mimic Kusto Explorer behavior when trying to run only comments.